core/setup.py

```diff
@@ -73,7 +73,7 @@
         # Accept patches but avoid automatically updating past a set minor version range.
         "dbt-extractor>=0.5.0,<=0.6",
         "minimal-snowplow-tracker>=0.0.2,<0.1",
-        "dbt-semantic-interfaces>=0.5.0,<0.6",
+        "dbt-semantic-interfaces>=0.5.1,<0.6",
```

Comment:
It is backwards compatible! Basically, I want to ensure that this feature is available for all users on core 1.8 once that gets released. We `could` leave the existing version range, but it might cause confusion for users if they somehow end up on DSI 0.5.0 and wonder why this feature isn't available!

Reply:
In this case it looks like a feature was added in the patch version bump to `0.5.1` from `0.5.0`, specifically a feature allowing metrics to be referenced in filter attributes of semantic layer objects. Arguably, this should have constituted a `0.6.0`, but the release of `0.5.1` has already happened. This makes `0.5.1` not explicitly backward compatible with `0.5.0` in that if someone were to specify a metric in a filter, parsing would break if `0.5.0` is the installed DSI version.

Reply:
So we'll want to get this change in prior to the final release of core 1.8.0

Reply:
> Arguably, this should have constituted a `0.6.0`, but the release of `0.5.1` has already happened. 

This makes sense.  As long as in future we try to follow semver more closely I'll ✅ this.  Since 1.7 is on 0.4 anyways it's low risk.

Reply:
Just me over her conflating backwards compatibility with being a purely additive change - sorry for the confusion!
And yes @QMalcolm point taken - we can do some work on the SL side to clean up our OSS versioning story in the future 🙏

Reply:
Thanks for the quick review - would one of y'all mind merging? I don't have permission.